hs_swat_modelinstance/models.py

```diff
@@ -1,79 +1,32 @@
 __author__ = 'Mohamed Morsy'
+from lxml import etree
+
 from django.contrib.contenttypes import generic
-from django.contrib.contenttypes.models import ContentType
-from django.core.exceptions import ValidationError, ObjectDoesNotExist
-from django.contrib.auth.models import User, Group
 from django.db import models
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
 
-from mezzanine.pages.models import Page, RichText
-from mezzanine.core.models import Ownable
 from mezzanine.pages.page_processors import processor_for
 
-from lxml import etree
-
 from hs_core.models import BaseResource, ResourceManager, resource_processor, CoreMetaData, AbstractMetaDataElement
+from hs_core.hydroshare import utils
 
+from hs_model_program.models import ModelProgramResource
+from hs_modelinstance.models import ModelInstanceMetaData, ModelOutput, ExecutedBy
 
+class ModelOutput(ModelOutput):
+    class Meta:
+        proxy = True
 
+class ExecutedBy(ExecutedBy):
+    class Meta:
+        proxy = True
 
 # extended metadata elements for SWAT Model Instance resource type
-class ModelOutput(AbstractMetaDataElement):
-    term = 'ModelOutput'
-    includes_output = models.BooleanField(default=False)
-
-    @classmethod
-    def create(cls, **kwargs):
-        return ModelOutput.objects.create(**kwargs)
-
-    @classmethod
-    def update(cls, element_id, **kwargs):
-        model_output = ModelOutput.objects.get(id=element_id)
-        if model_output:
-            for key, value in kwargs.iteritems():
-                setattr(model_output, key, value)
-
-            model_output.save()
-        else:
-            raise ObjectDoesNotExist("No ModelOutput element was found for the provided id:%s" % kwargs['id'])
-
-    @classmethod
-    def remove(cls, element_id):
-        raise ValidationError("ModelOutput element of a resource can't be deleted.")
-
-
-class ExecutedBy(AbstractMetaDataElement):
-    term = 'ExecutedBY'
-    name = models.CharField(max_length=200)
-    url = models.URLField()
-
-    def __unicode__(self):
-        self.name
-
-    @classmethod
-    def create(cls, **kwargs):
-        return ExecutedBy.objects.create(**kwargs)
-
-    @classmethod
-    def update(cls, element_id, **kwargs):
-        executed_by = ExecutedBy.objects.get(id=element_id)
-        if executed_by:
-            for key, value in kwargs.iteritems():
-                setattr(executed_by, key, value)
-
-            executed_by.save()
-        else:
-            raise ObjectDoesNotExist("No ExecutedBy element was found for the provided id:%s" % kwargs['id'])
-
-    @classmethod
-    def remove(cls, element_id):
-        raise ValidationError("ExecutedBy element of a resource can't be deleted.")
-
-
 class ModelObjectiveChoices(models.Model):
     description = models.CharField(max_length=300)
 
     def __unicode__(self):
-        self.description
+        return unicode(str(self))
```

Comment:
@mohamedmorsy You need to also override the __str__ method:

    def __str__(self):
        return "This is my string representation"

This way, the right thing will happen no matter if the code (e.g. Django) is running under Python 2.7 (where the call could be either str(myModelObjectiveChoices) or unicode(myModelObjectiveChoices)) or if code running under Python 3.5 (where the call can only be str(myModelObjectiveChoices)).